contracts/rewards/SimpleRewarder.sol

```diff
@@ -0,0 +1,229 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "@boringcrypto/boring-solidity-e06e943/contracts/libraries/BoringERC20.sol";
+import "@boringcrypto/boring-solidity-e06e943/contracts/libraries/BoringMath.sol";
+import "@boringcrypto/boring-solidity-e06e943/contracts/BoringOwnable.sol";
+import "../interfaces/IRewarder.sol";
+
+interface IMiniChef {
+    function lpToken(uint256 pid) external view returns (IERC20 _lpToken);
+}
+
+/**
+ * @title SimpleRewarder
+ * @notice Rewarder contract that can add one additional reward token to a specific PID in MiniChef.
+ * Emission rate is controlled by the owner of this contract, independently from MiniChef's owner.
+ * @author @0xKeno @weeb_mcgee
+ */
+contract SimpleRewarder is IRewarder, BoringOwnable {
+    using BoringMath for uint256;
+    using BoringMath128 for uint128;
+    using BoringERC20 for IERC20;
+
+    uint256 private constant ACC_TOKEN_PRECISION = 1e12;
+
+    /// @notice Info of each Rewarder user.
+    /// `amount` LP token amount the user has provided.
+    /// `rewardDebt` The amount of Reward Token entitled to the user.
+    struct UserInfo {
+        uint256 amount;
+        uint256 rewardDebt;
+    }
+
+    /// @notice Info of the rewarder pool
+    struct PoolInfo {
+        uint128 accToken1PerShare;
+        uint64 lastRewardTime;
+    }
+
+    /// @notice Address of the token that should be given out as rewards.
+    IERC20 public rewardToken;
+
+    /// @notice Var to track the rewarder pool.
+    PoolInfo public poolInfo;
+
+    /// @notice Info of each user that stakes LP tokens.
+    mapping(address => UserInfo) public userInfo;
+
+    /// @notice Total emission rate of the reward token per second
+    uint256 public rewardPerSecond;
+    /// @notice Address of the lp token that should be incentivized
+    IERC20 public masterLpToken;
+    /// @notice PID in MiniChef that corresponds to masterLpToken
+    uint256 public pid;
+
+    /// @notice MiniChef contract that will call this contract's callback function
+    address public immutable MINICHEF;
+
+    event LogOnReward(
+        address indexed user,
+        uint256 indexed pid,
+        uint256 amount,
+        address indexed to
+    );
+    event LogUpdatePool(
+        uint256 indexed pid,
+        uint64 lastRewardTime,
+        uint256 lpSupply,
+        uint256 accToken1PerShare
+    );
+    event LogRewardPerSecond(uint256 rewardPerSecond);
+    event LogInit(
+        IERC20 indexed rewardToken,
+        address owner,
+        uint256 rewardPerSecond,
+        IERC20 indexed masterLpToken
```

Comment:
Nitpicking: `rewardToken` and `masterLpToken` look like something that never changes so they do not necessarily need to be indexed.